### PR TITLE
Allow web's SafeAreaView to forward ref to its underlying view

### PR DESCRIPTION
**Pull request by antoine-cottineau**
## Summary

This PR solves the issue #295 by adding the ability for the web's `SafeAreaView` to forward the ref to the underlying `View`.

## Test Plan

I run the code by creating a basic Expo app and patching it with my changes.

### CODE REVIEW

**Review by janicduplessis — CHANGES_REQUESTED** [review]
Looks good, can you just fix the formatting in SafeAreaView.web.tsx?

**Comment by antoine-cottineau** [comment]
It should be good now 🙂

File: src/SafeAreaView.web.tsx [hunk 1]
```diff
@@ -19,67 +23,66 @@ const edgeBitmaskMap: Record<Edge, number> = {
...
 };
 
-export function SafeAreaView({
-  style = {},
-  mode,
-  edges,
-  ...rest
-}: NativeSafeAreaViewProps) {
-  const insets = useSafeAreaInsets();
+export const SafeAreaView: React.ForwardRefExoticComponent<
```
Comment (jacobp100) [inline src/SafeAreaView.web.tsx:26]: Do we need to annotate this type? It should be inferred from forwardRef, and might make the line breaking less aggressive

File: src/SafeAreaView.web.tsx [hunk 2]
```diff
@@ -19,67 +23,66 @@ const edgeBitmaskMap: Record<Edge, number> = {
...
+  NativeSafeAreaViewProps & React.RefAttributes<NativeSafeAreaViewInstance>
+> = React.forwardRef<NativeSafeAreaViewInstance, NativeSafeAreaViewProps>(
+  ({ style = {}, mode, edges, ...rest }, ref) => {
```
Comment (jacobp100) [inline src/SafeAreaView.web.tsx:29]: If we take `props` here and destructure it inside the component body, does it avoid the extra indentation?

**Review by jacobp100 — APPROVED** [review]
Looks good! Just a few things that might make the formatting less indented

File: src/SafeAreaView.web.tsx (hunk 1 above)
Comment (antoine-cottineau) [reply src/SafeAreaView.web.tsx:26]: Yes it removes a level of indentation! I thought we wanted to have the type explicitly written like in `SafeAreaView.tsx`.

File: src/SafeAreaView.web.tsx (hunk 2 above)
Comment (antoine-cottineau) [reply src/SafeAreaView.web.tsx]: Not needed anymore as removing the return type was enough to remove the extra indentation.

File: src/SafeAreaView.web.tsx (hunk 1 above)
Comment (janicduplessis) [reply src/SafeAreaView.web.tsx]: Can you also remove it from SafeAreaView.tsx for consistency?

Comment (antoine-cottineau) [reply src/SafeAreaView.web.tsx]: Yes! Done 😉

**Review by janicduplessis — APPROVED** [review]
Looks good! Thanks for fixing this

**Comment by antoine-cottineau** [comment]
My pleasure 🙂

**Comment by janicduplessis** [comment]
I might have messed something up, I published 4.5.3 which seems fine. Thanks for finding this.

**Pull request merged**